app/controllers/concerns/blueprints_filters.rb

```diff
@@ -88,10 +88,50 @@ def filter(blueprints)
 
       if @filters[:order] == "recent"
         blueprints = blueprints.reorder(created_at: :desc)
+      elsif @filters[:order] == "trending"
+        # Get blueprints from the last 30 days and calculate trending score
+        trending_query = <<-SQL
+          WITH blueprint_scores AS (
+            SELECT blueprints.id,
+              (
+                -- Recent engagement (last 7 days weighted more heavily)
+                (
+                  COALESCE((SELECT COUNT(*) FROM votes WHERE votes.votable_id = blueprints.id AND votes.created_at >= '#{7.days.ago}'), 0) * 2.0 +
+                  COALESCE((SELECT COUNT(*) FROM comments WHERE comments.blueprint_id = blueprints.id AND comments.created_at >= '#{7.days.ago}'), 0) * 3.0 +
+                  blueprints.usage_count * 4.0  -- Using usage_count directly
+                ) * 2.0 +  -- Double weight for recent activity
+                -- Total engagement
+                (blueprints.cached_votes_total * 1.0 + 
+                 COALESCE((SELECT COUNT(*) FROM comments WHERE comments.blueprint_id = blueprints.id), 0) * 2.0 +
+                 blueprints.usage_count * 3.0)
+              ) * 
+              -- Time boost: newer blueprints get a small boost
+              (1.0 + (1.0 - (EXTRACT(EPOCH FROM (blueprints.created_at - '#{30.days.ago}'::timestamp)) / 2592000.0)) * 0.5)
+              as trending_score
+            FROM blueprints
+            WHERE blueprints.created_at >= '#{30.days.ago}'
+          )
+          SELECT blueprints.*, COALESCE(blueprint_scores.trending_score, 0) as trending_score
+          FROM blueprints
+          LEFT JOIN blueprint_scores ON blueprint_scores.id = blueprints.id
+        SQL
+
+        blueprints = blueprints.from("(#{trending_query}) AS blueprints")
+          .includes(:mod, :tags, :tag_taggings, :user)
+          .order('trending_score DESC')
```

Comment:
What do you think about moving this to the model layer instead of keeping it in a controller concern? None of the controllers should be concerned with how the query is made and I would like to start setting a better precedent for keeping SQL things only in the model layer.

```suggestion
      blueprints.trending
```

where `trending` is a class method on the `Blueprint` model. Example:
```
  def self.trending
    # your code goes here
  end
```